File: app/src/main/java/com/hfad/pet_scheduling/data/local/entities/ScheduleTask.kt
package com.hfad.pet_scheduling.data.local.entities

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "schedule_tasks",
    foreignKeys = [
        ForeignKey(
            entity = Pet::class,
            parentColumns = arrayOf("petId"),
            childColumns = arrayOf("petId"),
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index(value = ["petId"])]
)
data class ScheduleTask(
    @PrimaryKey
    val taskId: String = java.util.UUID.randomUUID().toString(),

    val petId: String,
    val title: String,
    val description: String?,
    val category: String,
    val startTime: Long,
    val recurrencePattern: String,
    val recurrenceInterval: Int = 1,
    val reminderMinutesBefore: Int = 15,
    val isActive: Boolean = true,
    val createdAt: Long = System.currentTimeMillis(),
    val createdByUserId: String
)

File: PetSchedulingiOS/PetSchedulingiOS/Models/ScheduleTask.swift
//
//  ScheduleTask.swift
//  PetSchedulingiOS
//

import Foundation

struct ScheduleTask: Identifiable, Codable, Hashable {
    var id: String { taskId }
    let taskId: String
    let petId: String
    var title: String
    var description: String?
    var category: String
    var startTime: Int64
    var recurrencePattern: String
    var recurrenceInterval: Int
    var reminderMinutesBefore: Int
    var isActive: Bool
    let createdAt: Int64
    let createdByUserId: String
    
    init(taskId: String = UUID().uuidString, petId: String, title: String, description: String? = nil,
         category: String, startTime: Int64, recurrencePattern: String, recurrenceInterval: Int = 1,
         reminderMinutesBefore: Int = 15, isActive: Bool = true,
         createdAt: Int64 = Int64(Date().timeIntervalSince1970 * 1000),
         createdByUserId: String) {
        self.taskId = taskId
        self.petId = petId
        self.title = title
        self.description = description
        self.category = category
        self.startTime = startTime
        self.recurrencePattern = recurrencePattern
        self.recurrenceInterval = recurrenceInterval
        self.reminderMinutesBefore = reminderMinutesBefore
        self.isActive = isActive
        self.createdAt = createdAt
        self.createdByUserId = createdByUserId
    }
}
